pallets/identity/src/keys.rs

```diff
@@ -417,6 +417,10 @@ impl<T: Config> Module<T> {
             // Not really needed unless we allow identities to be deleted.
             Self::ensure_id_record_exists(target_did)?;
 
+            // Ensure we won't have too many keys.
+            let record = <DidRecords<T>>::get(target_did);
+            ensure_length_ok::<T>(record.secondary_keys.len().saturating_add(1))?;
+
             // Link the secondary key.
             Self::ensure_key_did_unlinked(&key)?;
```

Comment:
I think it would be useful to pull these two checks into one function `ensure_add_secondary_key(did, key)`.